src/main/java/com/surrealdb/refactor/driver/WsPlaintextConnection.java

```diff
@@ -81,15 +79,13 @@ public QueryBlockResult query(String query, List<Param> params) {
                                 }
                                 JsonElement outerResultJson = resp.get("result");
                                 QueryResult[] processedOuterResults;
-								
-                                
+                                ResultParser resultParser = new ResultParser();
```

Comment:
We can have the result parser as a class property. That way it can be injected for "production" with a correct implementation, but have a "fake" implementation injected for tests, such as a mock implementation. It also reduces garbage collection.

Reply:
I put it as a class property